pkg/kubevirt/core/data_volume.go

```diff
@@ -0,0 +1,106 @@
+package core
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	clouderrors "github.com/gardener/machine-controller-manager-provider-kubevirt/pkg/kubevirt/errors"
+
+	cdi "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DataVolumeManager manages Kubevirt DataVolume operations.
+type DataVolumeManager interface {
+	CreateVolume(ctx context.Context, name, namespace string, dvSpec cdi.DataVolumeSpec) error
+	GetVolume(ctx context.Context, name, namespace string) (*cdi.DataVolume, error)
+	ListDataVolumes(ctx context.Context, namespace string) (*cdi.DataVolumeList, error)
+	DeleteVolume(ctx context.Context, name, namespace string) error
+}
+
+type defaultDataVolumeManager struct {
+	c client.Client
+}
+
+// NewDefaultDataVolumeManager creates a new default manager with a k8s controller-runtime based on the sent kubecomfig.
+func NewDefaultDataVolumeManager(kubeconfig string) (DataVolumeManager, error) {
+	if kubeconfig == "" {
+		return nil, errors.New("kubevirt cluster kubeconfig cannot be empty")
+	}
+
+	config, err := clientcmd.RESTConfigFromKubeConfig([]byte(kubeconfig))
```

Comment:
Also add in cluster config

Reply:
tbh I added that and deleted it :-)

Reply:
> this also needs to come up with a new rbac role for cdi access

This can be configured in the extension itself and forwarded